lib/DXIL/DxilModule.cpp

```diff
@@ -2081,4 +2111,157 @@ bool DxilModule::IsPrecise(const Instruction *inst) const {
     return false;
 }
 
+bool DxilModule::ShaderCompatInfo::Merge(ShaderCompatInfo &other) {
+  bool changed = DXIL::UpdateToMaxOfVersions(minMajor, minMinor, other.minMajor,
+                                             other.minMinor);
+  if ((mask & other.mask) != mask) {
+    mask &= other.mask;
+    changed = true;
+  }
+  uint64_t rawBefore = shaderFlags.GetShaderFlagsRaw();
+  shaderFlags.CombineShaderFlags(other.shaderFlags);
+  if (rawBefore != shaderFlags.GetShaderFlagsRaw())
+    changed = true;
+  return changed;
+}
+
+void DxilModule::ClearShaderCompatInfo() { m_FuncToShaderCompat.clear(); }
+
+void DxilModule::ComputeShaderCompatInfo() {
+  // Initialize worklist with functions with callers
+  SmallSetVector<llvm::Function *, 8> worklist;
+
+  for (auto &function : GetModule()->getFunctionList()) {
+    if (function.isDeclaration() && !function.isIntrinsic() &&
+        function.getLinkage() ==
+            llvm::GlobalValue::LinkageTypes::ExternalLinkage &&
+        OP::IsDxilOpFunc(&function)) {
+      // update min shader model and shader stage mask per function
+      UpdateFunctionToShaderCompat(&function);
+    } else if (!function.isDeclaration()) {
+      // Initialize worklist with functions with callers.
+      // only used for validator version 1.8+
+      if (DXIL::CompareVersions(m_ValMajor, m_ValMinor, 1, 8) >= 0 &&
+          !function.user_empty())
+        worklist.insert(&function);
+
+      // Skip functions with shader flags already set
+      if (m_FuncToShaderCompat.count(&function) != 0)
```

Comment:
Yes, we still want to add it to the worklist, as another function that doesn't have flags set may call this one.  We go from callee to caller when combining flags.

Reply:
I'm removing this because I realize it could cause functions to skip flag collection, if an intrinsic used in the function was iterated first.  That also means I'm not bothering with retaining results from a prior run, so I'll clear the map each time this is called and get rid of the `ClearShaderCompatInfo` function as well.